### PR TITLE
perf(v2): smaller bundlesize by embedding metadata to content

### PR DESCRIPTION
**Pull request by endiliey**
## Motivation

This is breaking for those who swizzled Doc or Blog component.

For example, if you've swizzled DocItem. You'll have to update
```diff
- const {metadata, content: DocContent} = props;
+ const {content: DocContent} = props;
+ const {metadata} = DocContent;
```

Instead of creating a new chunk for metadata, metadata (usually <1kb) is now part of the content itself. The benefit is that we have much less number of request, smaller bundle size, less number of files to dispose during hot reload and we can scale better.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

Hot reload still working for blog and docs
blog hot reload still working
hot-reload-still-working
wds debug no unnecessary reload

Bundle size comparison

**Before**
First build 70.11kb, second build (cutting new version) 71.89kb, third build (cut another new version) 73.63kb. So roughly 1.74kbs per version

first bundle size  70 11kb
second bundle size  71 89kb
third bundle size  73 63kb

**After**
First build 67.35kb, second build (cut new version) 68.41kb, third build (cut new version) 69.5kb
So roughly 1kb per new version.

first bundle size 67 35kb
second bundle size 68 41kb
third bundle size 69 5kb

## TL;DR previously every new version cut cost us 1.74kb gzipped, now its only 1kb gzipped 😉 
Overall bundle size is also smaller :wew

### CODE REVIEW

File: packages/docusaurus-utils/src/index.ts [hunk 1]
```diff
@@ -27,7 +27,17 @@ export async function generate(
     return;
   }
 
-  const lastHash = fileHash.get(filepath);
+  let lastHash = fileHash.get(filepath);
+
+  // If file already exist, we try to calculate the content hash and compare
+  // This is to avoid unnecessary overwrite and we can reuse old file
+  if (!lastHash && fs.existsSync(filepath)) {
```
Comment (endiliey) [inline packages/docusaurus-utils/src/index.ts]: Oops maybe this should've been in separate PR

Comment (endiliey) [reply packages/docusaurus-utils/src/index.ts]: My dev reload (second one) went down from 7s to 5s because of this
